test/diff/print.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import "fmt"
+
+// Print takes a diff string generated by cmp.Diff and returns it
+// in a consistent format for reuse across all of our tests.
+func Print(diff string) string {
```

Comment:
one detail that might be important to mention the docstring or maybe even in the function name is that this expects that cmp.Diff has been called with the arguments in a specific order (want vs. got)

Reply:
another option is to wrap cmp.diff entirely and also display the error - if you use https://golang.org/pkg/testing/#B.Helper you'll get very similar functionality to what we have now, e.g. something in the test like

```
diff.FailIfWantNotGot(tc.expectedTaskSpec, *ts, "Didn't get expected TaskSpec modifications")
```

And that function would actually call Errorf

This is straying dangerously into non-go-like territory tho, your way is clearer! I think it's clarify vs ease

Reply:
I did consider wrapping cmp.Diff and I do think that is a very good idea.  Two considerations that lead me away from doing that here:

1. Some tests use `t.Fatalf` while others use `t.Errorf`.  So I figured a diff wrapping func would probably need to account for that too.
2. With an 80 file diff I wanted to do the very minimum to make all the various call sites uniform and limit the amount of review required.

I figure we can write a wrapping func for cmp.Diff in a future PR and then progressively replace cmp.Diff usage, perhaps over the course of multiple PRs?  Excellent point about B.Helper, I did not know about it!

Reply:
I really like the idea of a func with a name that points out the ordering ala `...WantGot(want, got)`; great idea!

Reply:
Create https://github.com/tektoncd/pipeline/issues/2587 to follow up.

Reply:
OK, I've updated the PR to use `diff.PrintWantGot` as the name and I've updated the docstring to make explicit the expectation of ordering.